doc/source/reference/optimize.rst

```diff
@@ -2,9 +2,3 @@
    :no-members:
    :no-inherited-members:
    :no-special-members:
-
-.. toctree::
-   :hidden:
-   :maxdepth: 1
-
-   optimize.cython_optimize
```

Comment:
This is now available in the `optimize` page as:

![image](https://user-images.githubusercontent.com/6570539/218624470-5eac3acc-5e78-4059-8125-eda8bdb9126a.png)

and

![image](https://user-images.githubusercontent.com/6570539/218624499-2c57f1f1-4bdf-4002-92e9-dc44bea31892.png)

It's not super prominent, but maybe that's OK. Should it be listed along with `show_options`, `OptimizeResult`, and `OptimizeWarning` at the top? It looks like the choice to make the `toctree` hidden was intentional. Fine with me either way.